### PR TITLE
fix a couple any types

### PR DESCRIPTION
**Pull request by fhennig**
resolves #

preview URL:

### PR Checklist

- [ ] All necessary documentation has been adapted.
- [ ] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: website/src/components/DataUseTerms/EditDataUseTermsModal.tsx [hunk 1]
```diff
@@ -47,8 +47,7 @@ type LoadedState = {
     earliestRestrictedUntil: DateTime | null;
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any -- TODO(#3451) use proper types
-function getLoadedState(rows: Record<string, any>[]): LoadedState {
+function getLoadedState(rows: Record<string, string | number | boolean | null>[]): LoadedState {
```
Comment (fengelniederhammer) [inline website/src/components/DataUseTerms/EditDataUseTermsModal.tsx:50]: This data is coming from the LAPIS details endpoint, isn't it? We already have a type for that:
```suggestion
function getLoadedState(rows: Details[]): LoadedState {
```
fromk `import type {Details} from "../../types/lapis.ts";`

File: website/src/utils/search.ts [hunk 2]
```diff
@@ -229,6 +228,7 @@ const makeCaseInsensitiveLiteralSubstringRegex = (s: string): string => {
     return `(?i)${s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}`;
 };
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (fengelniederhammer) [inline website/src/utils/search.ts:231]:
```suggestion
/* eslint-disable @typescript-eslint/no-explicit-any -- TODO(#3451) use proper types */
```

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
Thanks, nice improvement! I think we can improve a little more ;)

File: website/src/components/DataUseTerms/EditDataUseTermsModal.tsx (hunk 1 above)
Comment (fhennig) [reply website/src/components/DataUseTerms/EditDataUseTermsModal.tsx]: https://github.com/loculus-project/loculus/pull/3464/commits/b817828476b2acfab01801f968fe73cae9dca8ad

File: website/src/components/Navigation/DocsMenu.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { type FC } from 'react';
 import XIcon from '~icons/material-symbols/close';
 import MenuIcon from '~icons/material-symbols/menu';
 
-type Page = MDXInstance<Record<string, any>>; // eslint-disable-line @typescript-eslint/no-explicit-any -- TODO(#3451) use a proper type
+type Page = MDXInstance<Record<string, unknown>>;
```
[the remark this answers is not in the document]
Comment (fhennig) [reply website/src/components/Navigation/DocsMenu.tsx]: I reverted the change.

Looking "upstream" where the pages are coming from, they just come in as 'any' and I thought it wouldn't be very helpful to cast there, without actually validating. And I didn't want to validate now (I just wanted to do a few quick changes)

I thought 'unknown' is fine in the sense that we do not know what the frontmatter looks like. We'd have to properly validate (and come up with what to do if i.e. 'title' is missing). Maybe doing this one is for another PR 😄

**Review by fengelniederhammer — APPROVED** [review]

File: website/src/components/Navigation/DocsMenu.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (fengelniederhammer) [reply website/src/components/Navigation/DocsMenu.tsx]: Sounds good to me. At least its obvious that this part needs improvemenets :+1:

**Pull request merged**